server/backend/src/users/users.service.ts

```diff
@@ -29,6 +31,42 @@ export class UsersService {
     return user.groups;
   }
 
+  /**
+   * Register a camera to the given user and group
+   * 
+   * @param userId the user's ID.
+   * @param groupId the group ID.
+   * @param name New camera's name
+   * @param pubKey New camera's public key
+   * @param serial New camera's serial
+   */
+  public async putCamera(
+    userId: string,
+    groupId: string,
+    name: string,
+    pubKey: string,
+    serial: string
+  ): Promise<Camera> {
+    const group = await this.groupRepository.findOne(
+      { id: groupId },
+      { populate: ['user', 'cameras']},
+    );
+
+    if (group === null) {
+      throw new NotFoundError(`Group with ID "${groupId}" does not exist.`);
+    }
+
+    if (group.user.id !== userId) {
+      throw new UnauthorizedException();
+    }
```

Comment:
If we grab the user rather than the group, we can avoid directly throwing an `UnauthorizedException` directly:
```suggestion
    const user = await this.userRepository.findOne(
      { id: userId },
      { populate: ['groups', 'groups.cameras']},
    );

    const groups = user.groups.matching({ where: { id: groupId }, limit: 1 });
    if (groups.count === 0) {
      throw new NotFoundError(`Group with ID "${groupId}" does not exist.`);
    }
    const group = groups[0];
```
I'm not exactly sure on the parameter for [`Collection.matching()`](https://mikro-orm.io/docs/collections#filtering-collections) (no VSCode typehints for me at school), but I would imagine we can add a `where` clause.

Reply:
I was looking at the matching function last night. It looks like it sorts the collection then returns the top `limit`.
https://mikro-orm.io/docs/collections#filtering-collections
I didn't put the where clause so maybe I can try again. Would you rather not return `UnauthorizedException`? This seemed simpler then getting the user then going through the list.

Reply:
I thought about it more and not returning an `UnauthorizedException` would be useful if we were doing unit tests (which Mikro-ORM makes impoSSIBLE AAAAAAA), so it's probably fine to return it for now, and if we ever rip MikroORM out, we can refactor later.

Reply:
We could return NotFound instead, similar to what we do for the other stuff.

Reply:
That sounds good